gnowsys-ndf/gnowsys_ndf/ndf/templatetags/ndf_tags.py

```diff
@@ -1206,10 +1206,10 @@ def get_prior_node(node_id):
 	if topic_GST._id in obj.member_of:
 
 		if obj.prior_node:
-			for each in obj.prior_node:
-				node = node_collection.one({'_id': ObjectId(each) })
-				prior.append(( node._id , node.name ))
-
+	#	for each in obj.prior_node:
+#			node = node_collection.one({'_id': ObjectId(each) })
+	#		prior.append(( node._id , node.name ))
+			prior=[(node_collection.one({'_id': ObjectId(each) })._id,node_collection.one({'_id': ObjectId(each) }).name) for each in obj.prior_node]
```

Comment:
While changing the loops into List Comprehension form,we encountered this situation only at this particular loop.Here the database query is on '_id' which is indexed by default. so we took this step.